お部屋の内見一覧ページをデザインする

uchihiro04 opened the pull request:
## 概要 
Tailwind CSSを用いて、お部屋の内見一覧ページにデザインを入れました。また、以下の変更を行いました。
* スコア入力ページへのリンクについて、リンクの範囲を行全体に変更

## スクリーンショット
### お部屋の内見一覧ページ
<img width="600" alt="230712_内見一覧ページ" src="https://github.com/uchihiro04/house-viewing-score-log/assets/77523896/b83622d5-c4fc-4063-b0a6-a424ef62ecda">

デベロッパツールで「iPhone SE」の画面（375 × 667）を想定して表示した場合
<img width="200" alt="230712_内見一覧ページスマートフォン" src="https://github.com/uchihiro04/house-viewing-score-log/assets/77523896/0b1ca684-aa01-4527-be9c-2d973bb17c47">

### フラッシュメッセージが表示された場合
<img width="600" alt="230712_内見一覧ページフラッシュメッセージ" src="https://github.com/uchihiro04/house-viewing-score-log/assets/77523896/4b2b8ab6-97a4-4b74-865f-0faff259495c">

デベロッパツールで「iPhone SE」の画面（375 × 667）を想定して表示した場合
<img width="200" alt="230712_内見一覧ページフラッシュメッセージ_スマホ" src="https://github.com/uchihiro04/house-viewing-score-log/assets/77523896/9562fcd6-5b79-48b8-b8ea-c494e05af0ad">

## 関連Issue
- #102 

Close #102

uchihiro04 committed:
#102 不要なCSSを削除

アプリケーションをデザインしていく上で不要なCSSを削除しました。

uchihiro04 committed:
#102 見出しのデザインを実装

uchihiro04 committed:
102 テーブルのレイアウトを実装

テーブルのレイアウトを実装しました。中身の文字の配置などは未実装です。

uchihiro04 committed:
102 スコア入力数の位置を変更

スコア入力数の位置を、部屋名の下に変更しました。

uchihiro04 committed:
102 お部屋一覧に矢印のアイコンを追加

お部屋一覧に矢印のアイコンを追加しました。現時点ではクリック・タップしても何も起こりません。

uchihiro04 committed:
102 テーブルの幅を調整

部屋名の文字数が少ない時にテーブルの幅が狭くならないよう調整しました。

uchihiro04 committed:
#102 説明文・テーブルのレイアウトを調整

説明文がテーブルと同じ幅で折り返すように調整しました。また、パソコンで見る際のテーブルの大きさを調整しました。

uchihiro04 committed:
#102 Slimの文字数に関するルールを無効化

Slimの文字数について、Tailwind CSSを使うことで文字数が長くなるためルールを無効化しました。

uchihiro04 committed:
#102 「スコアを見る」ボタンのデザインを設定

uchihiro04 committed:
102 矢印の位置を変更

お部屋一覧の矢印について、スマートフォンとパソコンで大きく位置が変わらないように調整しました。

uchihiro04 committed:
#102 スコア未入力時の「スコアを見る」ボタンのデザインを設定

スコアが全員未入力の時、「スコアを見る」ボタンがグレーアウトする形に設定しました。

uchihiro04 committed:
102 リンクの範囲が行全体になるように設定

スコア入力ページへのリンクについて、リンクの範囲を行全体に設定しました。

uchihiro04 committed:
#102 フラッシュメッセージのデザインを設定

フラッシュメッセージ（notice・alert）のそれぞれに、デザインを設定しました。

Yuma-Nagaoka commented on app/views/house_viewings/rooms/index.html.slim:
```diff
@@ -1,15 +1,26 @@
-= link_to_if Room.score_entered?(@rooms.ids), 'スコアを見る', house_viewing_scores_path
+  p.md:w-1/3.mx-auto 1件以上スコアの入力が完了していると、スコアを見ることができます。
+  - if Room.score_entered?(@rooms.ids)
+    = link_to 'スコアを見る', house_viewing_scores_path, class: 'bg-slate-950 block hover:bg-slate-700 mb-5 py-2 px-4 rounded text-center text-white md:w-1/3 mx-auto'
```
やはりclassが多くて見づらく感じるため、cssにbtnクラスを作って
`bg-slate-950 block hover:bg-slate-700 mb-5 py-2 px-4 rounded text-center text-white md:w-1/3 mx-auto`
をapplyし、
`button.bg-slate-950.hover:bg-slate-700.mb-5.py-2.px-4.rounded.text-center.text-white.w-full.md:w-1/3.mx-auto[data-action="clipboard#copySharedUrl"] 招待URLをコピー`
の部分でも流用しちゃいましょう

uchihiro04 replied:
見た目を整えようとしてクラスが増えた結果、コードの可読性が下がっていますね……。

いただいたレビューをもとに、ボタンに関するクラスでまとめられる箇所を共通化しました。
別の箇所で小さいボタンが必要になるため、クラス名は`btn-lg`としています。該当するコミットは[98d28a1](https://github.com/uchihiro04/house-viewing-score-log/pull/134/commits/98d28a1b8476025d485a7575f11b0447974383f0)です。

Yuma-Nagaoka commented on app/views/layouts/application.html.slim:
```diff
@@ -10,6 +10,11 @@ html
-      = flash.notice.presence || flash.alert.presence
+    main
+      - if flash.notice.present?
+        .bg-blue-100.border-t.border-b.border-blue-500.text-blue-700.px-4.py-3 role="alert"
```
ここも共通のcss部分をclass化しちゃいましょう

uchihiro04 replied:
いただいたレビューをもとに、フラッシュメッセージに関するクラスをまとめ、新しいクラスとして定義しました。noticeとalertで色が異なるため、色に関する部分はまとめておりません。
該当するコミットは[3ed744a](https://github.com/uchihiro04/house-viewing-score-log/pull/134/commits/3ed744aa097d8ca21cd01e14e3d94cfeb0c21bd8)です。

Yuma-Nagaoka commented on app/views/house_viewings/rooms/index.html.slim:
```diff
@@ -1,15 +1,26 @@
-        td = "スコア入力数: #{room.scores.length}"
+header.border-b.border-black
+  .container.flex.justify-center.mx-auto.p-5
+    h1.font-bold.text-2xl 内見したお部屋一覧
```
ここら辺のヘッダー部分やテーブル部分も他の画面と共通だと思うので、このPRでなくてもいいですが後で共通のcss classにしましょう

uchihiro04 replied:
ヘッダー部分に関しましてはどのページでも必ず存在するため、他のページを作成する前に今回のPRで共通のCSSクラスとして定義しました。
テーブルに関する部分につきましては、テーブルが出てくる別のIssueにて共通のクラスを作成したいと思います。
該当するコミットは[cceb9e0](https://github.com/uchihiro04/house-viewing-score-log/pull/134/commits/cceb9e06576636903180c49b9dea81ab823d15a2)です。

Yuma-Nagaoka commented on app/javascript/controllers/link_controller.js:
```diff
@@ -0,0 +1,8 @@
+import { Controller } from "@hotwired/stimulus"
+
+export default class extends Controller {
+  createLink(event) {
```
やっていることはurl遷移なのでcreateじゃないと思います

uchihiro04 replied:
ご指摘いただいた通り、メソッドの中身はURLの遷移であるためcreateLinkは不適切ですね……。メソッドを適用する箇所がリンクであることを意識しすぎてしまいました。
いただいたレビューをもとに、transitionUrl（URLを遷移する）メソッドに変更しました。また、ファイル名もlink_controller.jsからurl_controller.jsに変更しました。該当するコミットは[948d95d](https://github.com/uchihiro04/house-viewing-score-log/pull/134/commits/948d95d73678f14c21810cf1ea99e4e15deafe51)です。

uchihiro04 committed:
#102 ボタンに関するクラスをまとめる

ボタンに関するクラスをbtn-lgクラスとしてまとめました。

uchihiro04 committed:
#102 フラッシュメッセージに関するクラスを作成

フラッシュメッセージ（notice、alert）について共通化できるクラスをまとめて、新しいクラスとして定義しました。

uchihiro04 committed:
#102 ヘッダーに関するクラスを作成

クラス名が長くならない形で他のページでも利用できるように、ヘッダーに関するクラスを新たに作成しました。

uchihiro04 committed:
#102 ファイル名・メソッド名の変更

URLの遷移に関するJSのファイルについて、link_controller.jsからurl_controller.jsに名前を変更しました。また、メソッド名もcreateLinkからtransitionUrlに変更しました。

Yuma-Nagaoka commented on app/javascript/controllers/url_controller.js:
```diff
@@ -0,0 +1,8 @@
+import { Controller } from "@hotwired/stimulus"
+
+export default class extends Controller {
+  transitionUrl(event) {
```
redirectToUrlとかが良さそうですかね
transitionUrlだと名詞+名詞なのでfunctionとしては微妙そうです

Yuma-Nagaoka replied:
すみません自動詞でtransitionありました。
他動詞ではないのでtransition_to_urlが良さそうです
(あと、リダイレクト処理ではなかったですね・・)

uchihiro04 replied:
自動詞と他動詞に関しては意識できていなかったです……。
いただいたレビューをもとに、メソッド名をtransitionUrlからtransitionToUrlに変更しました。該当するコミットは[7a99024](https://github.com/uchihiro04/house-viewing-score-log/pull/134/commits/7a99024910a218f9fc6861570fd91e450da54c2e)です。

uchihiro04 committed:
#102 ファイル名・メソッド名の変更

URLの遷移に関するJavaScriptのメソッドについて、名前をtransitionUrlからtransitionToUrlに変更しました。

Yuma-Nagaoka reviewed (approved):
` .bg-blue-100.border-blue-500.text-blue-700.flash-message`
ここ微妙に共通化できないんですね
LGTMです

takuminasu commented on app/views/house_viewings/rooms/index.html.slim:
```diff
@@ -1,15 +1,26 @@
+  table.mt-6.mb-3.text-left.table-auto.w-full.md:w-1/3.mx-auto
+    tbody
+      - @rooms.each do |room|
+        tr.border.border-black.cursor-pointer[data-controller="url" data-action="click->url#transitionToUrl" data-href=new_house_viewing_room_score_path(room_id: room.id)]
```
link_toから変更した理由ってなんでしょうか？

uchihiro04 replied:
link_toから変更した理由は、クリック・タップしたときに遷移できるリンクの範囲を広げたかったためになります。

link_toを使っていた際は、部屋名をクリック・タップするとお部屋のスコア入力ページに遷移できる形でした。
スマートフォンで使うことを想定した場合、部屋名など行の一部がリンクの範囲になっているよりも、行全体にリンクの範囲を広げた方がタップしやすいと考えたため、link_toから変更して行全体がリンクになる形に変更しています。

takuminasu replied:
なるほど。

jsでどうこうよりcssでlinkの幅を広げるとかで対応したいですね。
このようなことを相談できると良かったと思います。

すみませんがここのデザインtableタグよりulタグの方が適切のような気がしますのでそちらでデザインを当て直してもらってもいいですか？
ベースは作成したのでボーダーとか余白の調整をしてもらえればと思います。

```ruby
.mx-10
  ul.md:w-1/3.mx-auto
    - @rooms.each do |room|
      li.p-2
        = link_to new_house_viewing_room_score_path(room_id: room.id) do 
          .flex.justify-between
            div
              div  
                = room.name
              div
                = "スコア入力数: #{room.scores.length}"
            div
              svg.w-4.h-4.mx-auto xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 8 14"
                path[stroke="currentColor" stroke-linecap="round" stroke-linejoin="round" stroke-width="1"
                d="m1 13 5.7-5.326a.909.909 0 0 0 0-1.348L1 1"]
```

uchihiro04 replied:
ご指摘いただき考え直したところ、列が1列しかなく列の見出しも存在しないため、テーブルよりリストのほうが適切だと思いました……。
また、リストにしたことで、JavaScriptを使わなくてもリンクの範囲を広げることができました。実装の方針に関しましては今後擦り合わせを行うように意識したいと思います。

いただいたレビューをもとに修正を行いました。該当するコミットは[ed640a1](https://github.com/uchihiro04/house-viewing-score-log/pull/134/commits/ed640a1a2c4701b57de0b709234047ff85644768)です。

uchihiro04 committed:
#102 お部屋一覧をテーブルからリストに変更

お部屋一覧について、テーブルからリストに変更しました。また、不要になったJavaScriptファイルを削除しました。

takuminasu reviewed (approved):
LGTM
